OniProj/OniGameSource/Oni_Level.c

```diff
@@ -1213,7 +1213,7 @@ IMtShade ONrLevel_ObjectGunk_GetShade(ONtObjectGunk	*inObjectGunk)
 	UUmAssert(g <= 0xFF);
 	UUmAssert(b <= 0xFF);
 
-	shade = (0xFF << 24) | (r << 16) | (g << 8) | (b << 0);
+	shade = ((IMtShade)0xFF << 24) | (r << 16) | (g << 8) | (b << 0);
```

Comment:
Would it be clearer to the reader if it used `UUtUns32` instead of `IMtShade`?
Then the reader immediately knows it is 32-bits.
Reading this line, I don't know what type and bit width `IMtShade` has.

Reply:
`shade` is of type `IMtShade`, so the operations should definitely be happening with a constant of type `IMtShade`.
It is correct that it's not immediately obvious it's a typedef for `UUtUns32`. That argument would apply to any `typedef` though, so if `UUtUns32` should be preferred then I think the `IMtShade` typedef should be eliminated entirely.
An alternative solution could be to introduce a function like `IMtShade MakeShadeARGB(UUtUns8 a, UUtUns8 r, UUtUns8 g, UUtUns8 b)`.

Reply:
I like this `MakeShadeARGB` proposal.